### PR TITLE
feat(datestrings): Allow string dates, closes #1765

### LINKED ISSUE #1765: JsonStreamWriter Support ISO Date String
What the problem is.
Currently DATE field [only accepts int values ](https://cloud.google.com/bigquery/docs/write-api#data_type_conversions)
is it possible to accept ISO date String too for example `1999-01-01`

What you want to happen.
being able to load date strings with JsonStreamWriter

### LINKED ISSUE #11975: [java-bigquerystorage] JsonStreamWriter Support ISO Date String
What the problem is.
Currently DATE field [only accepts int values ](https://cloud.google.com/bigquery/docs/write-api#data_type_conversions)
is it possible to accept ISO date String too for example `1999-01-01`

What you want to happen.
being able to load date strings with JsonStreamWriter

### PR DESCRIPTION
**Pull request by dark0dave**
Closes googleapis/google-cloud-java#11975 

Signed-off-by: dark0dave <dark0dave@mykolab.com>

### CODE REVIEW

**Comment by dark0dave** [comment]
@ismailsimsek will this do?

**Comment by ismailsimsek** [comment]
@dark0dave it looks good to me +1

**Comment by dark0dave** [comment]
Should be ready now. Apologies, its the holiday season.

**Comment by dark0dave** [comment]
@ismailsimsek let me know if you need any other types for dates to be added. Longs/shorts/floats etc.

**Comment by dark0dave** [comment]
@aribray this is ready for review please :), thank you.

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessage.java [hunk 1]
```diff
@@ -710,6 +716,12 @@ private static void fillRepeatedField(
           }
           break;
         case STRING:
+          if (fieldSchema != null && fieldSchema.getType() == TableFieldSchema.Type.DATE) {
+            if (val instanceof String) {
+              protoMsg.setField(fieldDescriptor, (String) val);
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessage.java:721]: Should be repeated field?

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessageTest.java [hunk 2]
```diff
@@ -731,11 +731,18 @@ public void testDate() throws Exception {
         TableSchema.newBuilder()
             .addFields(TableFieldSchema.newBuilder(TEST_DATE).setName("test_string").build())
             .addFields(TableFieldSchema.newBuilder(TEST_DATE).setName("test_long").build())
+            .addFields(TableFieldSchema.newBuilder(TEST_DATE).setName("test_iso_string").build())
+            .build();
+    TestDate expectedProto =
+        TestDate.newBuilder()
+            .setTestString(18935)
+            .setTestLong(18935)
+            .setTestIsoString("1999-01-01")
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessageTest.java:740]: Add a test for repeated field.

**Comment by yirutang** [comment]
> is there any update on this feature?

Waiting for the owner to address review feedbacks? If I didn't hear him for a week, I will submit a patched PR.

**Pull request closed without merging**